README.md

```diff
@@ -42,6 +42,19 @@ Your project will be versioned according to the latest tag found in the commit h
 
 When you want to release a version of your software, whether it's a pre-release, RTM, patch, or anything else, simply create a tag with a name which is a valid [SemVer 2.x](https://semver.org/spec/v2.0.0.html) version and build your projects. MinVer will apply the version to the assemblies and packages. (If you like to prefix your tag names, see the [FAQ](#can-i-prefix-my-tag-names).)
 
+Keep in mind that MinVer is unaware of the branch you are working on and does not possess knowledge about your release process. As a result, it is compatible with any branching or release strategy of your choice. Nonetheless, it is essential to **tag the specific commit on the branch being released**.
+
+For example, if you automatically start a release pipeline with every pull request to the `main` branch and tag the latest change with a new version before creating the pull request, you will still receive a new pre-release version number. This is because the current commit on the main branch is not the commit with your tag on it, as described in the ["How it works"](#how-it-works) section.
+
+The merge from the Pull Request creates a new commit on the `main` branch. To properly handle this, tag the merge commit on the `main` branch using a command like:
+
+```shell
+git tag 1.2.3
+git push --tags
+```
+
+Afterwards manually trigger the CI/CD pipeline to release the correct version.
+

```

Comment:
@rodyvansambeek what do you think of this?

```suggestion
When you want to release a version of your software, whether it's a pre-release, RTM, patch, or anything else, simply create a tag on the commit you want to release with a name which is a valid [SemVer 2.x](https://semver.org/spec/v2.0.0.html) version and build your projects. For example:

\```shell
git tag 1.2.3
git push --tags
\```

When you build your software from the commit with that tag, MinVer will apply the version to the assemblies and packages. (If you like to prefix your tag names, see the [FAQ](#can-i-prefix-my-tag-names).)

Bear in mind that MinVer is unaware of the branches which contain the commit, nor your release process, so [it is compatible with any branching or release strategy you choose](#can-i-use-my-own-branching-strategy).

```

(Note that this suggestion cannot be accepted as-is, because I had to escape the nested instances of ``` with `\`.)

I'd prefer not to get too far into examples of specific scenarios, since they are unbounded. I'd hope that everything in the text I'm proposing to remove flows naturally from what I'm proposing we leave in.

Reply:
I think that is better indeed. Clear, concise and to the point. I'll update the PR.